qdb/tracer.py

```diff
@@ -104,16 +132,18 @@ def _init(self, config=None, merge=False, **kwargs):
             self.log_handler = FileHandler(config.log_file)
             self.log_handler.push_application()
 
-        # The timing between these lines might matter depending on the
-        # cmd_manager. Don't seperate them.
-        self.cmd_manager = (config.cmd_manager or RemoteCommandManager)(self)
-        self.cmd_manager.start(config.auth_msg)
+        self.bound_cmd_manager = config.cmd_manager or TerminalCommandManager()
+        self.bound_cmd_manager.start(config.auth_msg)
 
         # We need to be able to send stdout back to the user debugging the
         # program. We hold a handle to this in case the program resets stdout.
+        self._old_stdout = sys.stdout
+        self._old_stderr = sys.stderr
+        self.redirect_output = (
```

Comment:
If you are using a terminal command manager, the output is "redirected" so it does what you want. I think I could maybe do a warning, but because it is still doing what you asked (a different way) I don't think it should be an exception.